google/cloud/aiplatform/tensorboard/uploader_utils.py

```diff
@@ -0,0 +1,377 @@
+                    projects/{project}/locations/{location}/tensorboards/{tensorboard}/experiments/{experiment}/runs/{run}
+            api (TensorboardServiceClient):
+                Required. A TensorboardServiceStub.
+        """
+        self._run_resource_id = run_resource_id
+        self._api = api
+        self._tag_to_time_series_proto: Dict[
+            str, tensorboard_time_series.TensorboardTimeSeries
+        ] = {}
+
+    def get_or_create(
+        self,
+        tag_name: str,
+        time_series_resource_creator: Callable[
+            [], tensorboard_time_series.TensorboardTimeSeries
+        ],
+    ) -> tensorboard_time_series.TensorboardTimeSeries:
+        """
+        Get a time series resource with given tag_name, and create a new one on
+        OnePlatform if not present.
+
+        Args:
+            tag_name (str):
+                Required. The tag name of the time series in the Tensorboard log dir.
+            time_series_resource_creator (Callable[[], tensorboard_time_series.TensorboardTimeSeries]):
+                Required. A callable that produces a TimeSeries for creation.
+
+        Returns:
+            time_series (tensorboard_time_series.TensorboardTimeSeries):
+                A new or existing tensorboard_time_series.TensorbaordTimeSeries.
+
+        Raises:
+            exceptions.InvalidArgument:
+                The tag_name or time_series_resource_creator is an invalid argument
+                to create_tensorboard_time_series api call.
+            ExistingResourceNotFoundError:
+                Could not find the resource given the tag name.
+            ValueError:
+                More than one time series with the resource name was found.
+        """
+        if tag_name in self._tag_to_time_series_proto:
+            return self._tag_to_time_series_proto[tag_name]
+
+        time_series = time_series_resource_creator()
+        time_series.display_name = tag_name
+        try:
+            time_series = self._api.create_tensorboard_time_series(
+                parent=self._run_resource_id, tensorboard_time_series=time_series
+            )
+        except exceptions.InvalidArgument as e:
+            # If the time series display name already exists then retrieve it
+            if "already exist" in e.message:
+                list_of_time_series = self._api.list_tensorboard_time_series(
+                    request=tensorboard_service.ListTensorboardTimeSeriesRequest(
+                        parent=self._run_resource_id,
+                        filter="display_name = {}".format(json.dumps(str(tag_name))),
+                    )
+                )
+
+                if not list_of_time_series:
+                    raise ExistingResourceNotFoundError(
+                        "Could not find time series resource with display name: {}".format(
+                            tag_name,
+                        )
+                    )
+
+                if len(list_of_time_series) > 1:
```

Comment:
`list_tensorboard_time_series` returns a pager: https://github.com/googleapis/python-aiplatform/blob/c795c6fbb87c4f71845cfbd2647c1adbc029bcef/google/cloud/aiplatform_v1beta1/services/tensorboard_service/pagers.py#L429

These usually don't support `len` and the `bool` usage above may not work as expected. It seems like it needs to be converted to a `list` after the API request or handled iteratively.